### PR TITLE
Added E2E Test: Client Credentials with Certificate from Key Vault

### PR DESCRIPTION
**Pull request by Robbie-Microsoft**
I successfully ran this e2e test to completion/success on my devbox with hardcoded values.

This PR only adds an e2e test. After this is merged, I would like to re-write the sample in TypeScript, then base the e2e test off of the TypeScript code - something we haven't done yet in the msal-node samples/e2e tests.

Additionally, some potential follow up work would be to re-use some of the certificate transforming functionality introduced here and change the way developers pass in certificates.

### CODE REVIEW

File: samples/msal-node-samples/client-credentials-with-cert-from-key-vault/test/client-credentials-with-cert-from-key-vault-aad.spec.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            LAB_KEY_VAULT_URL
+        );
+        [thumbprint, privateKey, x5c] = await getCertificateInfo(
+            keyVaultSecretClient,
+            LAB_CERT_NAME
+        );
+
+        // TODO // get these from Lab Client, or am I allowed to save these as constants?
+        config.authOptions.clientId = "microsoft client id";
+        config.authOptions.authority = "microsoft authority";
```
Comment (Robbie-Microsoft) [inline samples/msal-node-samples/client-credentials-with-cert-from-key-vault/test/client-credentials-with-cert-from-key-vault-aad.spec.ts:40]: CC @hectormmg I was unable to get these values from the LabClient. It returns different values than the ones that I want.

Can I hardcode these as constants, or should I try to get from the LabClient?

I'll ping you regarding the values I'm talking about.

File: samples/msal-node-samples/client-credentials-with-cert-from-key-vault/test/client-credentials-with-cert-from-key-vault-aad.spec.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        const keyVaultSecretClient = await getKeyVaultSecretClient(
+            LAB_KEY_VAULT_URL
+        );
+        [thumbprint, privateKey, x5c] = await getCertificateInfo(
+            keyVaultSecretClient,
+            LAB_CERT_NAME
+        );
+
+        config.authOptions.clientId = process.env.AZURE_CLIENT_ID;
+        config.authOptions.authority = `https://login.microsoftonline.com/${process.env.AZURE_TENANT_ID}`;
```
Comment (Robbie-Microsoft) [inline samples/msal-node-samples/client-credentials-with-cert-from-key-vault/test/client-credentials-with-cert-from-key-vault-aad.spec.ts]: TODO: I think I can get these values, and possibly the keyvault name and url from the lab client?

Comment (bgavrilMS) [reply samples/msal-node-samples/client-credentials-with-cert-from-key-vault/test/client-credentials-with-cert-from-key-vault-aad.spec.ts]: For client_credentails, I see that MSAL.NET hardcodes everything - https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/tests/Microsoft.Identity.Test.Integration.netcore/Infrastructure/ConfidentialAppSettings.cs#L40

It's ok to hardcode client id, tenant id, keyvault names - these are not secrets.

File: .pipelines/3p-e2e.yml [hunk 3]
```diff
@@ -46,7 +46,7 @@ resources:
         - repository: 1P
           type: git
           name: IDDP/msal-javascript-1p
-          ref: master
+          ref: rename_cert_path
```
Comment (Robbie-Microsoft) [inline .pipelines/3p-e2e.yml]: This ref and the rest of this file's changes will be reverted before merging into dev. This ref change is needed for pointing to the corresponding 1P change.

File: samples/e2eTestUtils/package.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
     "private": true,
     "main": "src/index.ts",
     "dependencies": {
-        "@azure/identity": "^4.3.0",
+        "@azure/identity": "^4.5.0-beta.3",
```
Comment (hectormmg) [inline samples/e2eTestUtils/package.json:8]: @Robbie-Microsoft I'd like to avoid betas if possible, do we have an ETA for the fix to GA?

Comment (Robbie-Microsoft) [reply samples/e2eTestUtils/package.json:8]: 4.5.0 ETA is next Tuesday

Comment (Robbie-Microsoft) [reply samples/e2eTestUtils/package.json:8]: Delayed til Thursday or next Tuesday. Will post updates here.

File: samples/e2eTestUtils/src/CertificateUtils.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+    // get the private key and the public certificate, in PKCS 12 format
+    const pkcs12Certificate = Buffer.from(PKCS12CertificateBase64, "base64");
+
+    // write the PKCS#12 certificate to a temporary file
+    fs.writeFileSync(p12FilePath, pkcs12Certificate);
+
+    try {
+        // get the private key from the pkcs12 file through openssl, via a synchronous child process
+        execSync(
```
Comment (gladjohn) [inline samples/e2eTestUtils/src/CertificateUtils.ts:35]: are there any node js libraries like pkcs12, or pem to handle PKCS#12 files directly in memory? why use openssl?

File: samples/e2eTestUtils/src/CertificateUtils.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    const PKCS12CertificateBase64: string = await getKeyVaultSecret(
+        client,
+        secretName
+    );
+
+    // get the private key and the public certificate, in PKCS 12 format
+    const pkcs12Certificate = Buffer.from(PKCS12CertificateBase64, "base64");
+
+    // write the PKCS#12 certificate to a temporary file
+    fs.writeFileSync(p12FilePath, pkcs12Certificate);
```
Comment (gladjohn) [inline samples/e2eTestUtils/src/CertificateUtils.ts:31]: can this be done in memory?

File: samples/e2eTestUtils/src/Constants.ts [hunk 7]
```diff
@@ -6,6 +6,8 @@ export const ENV_VARIABLES = {
 
 export const LAB_API_ENDPOINT = "https://msidlab.com/api";
 export const LAB_SCOPE = "https://request.msidlab.com/.default";
+export const LAB_KEY_VAULT_URL = "https://msidlabs.vault.azure.net";
+export const LAB_CERT_NAME = "LabAuth";
```
Comment (gladjohn) [inline samples/e2eTestUtils/src/Constants.ts:10]: doesn't MSAL JS already install this cert to the build machine? @hectormmg

File: samples/e2eTestUtils/package.json (hunk 4 above)
Comment (KarishmaGhiya) [reply samples/e2eTestUtils/package.json:8]: Hello, we released @azure/identity 4.5.0 GA today.

**Pull request closed without merging**